### PR TITLE
Add rake task for removing sub lists with no subs

### PR DESCRIPTION
**Pull request by SamJamCul**
https://trello.com/c/TcWzthQL/880-clean-subscriber-lists-created-by-finder-frontend

### CODE REVIEW

File: lib/tasks/clean.rake [hunk 1]
```diff
@@ -37,6 +37,19 @@ namespace :clean do
...
   end
 
+  desc "Remove SubscriberLists with no subscriptions"
+  task remove_empty_subscriberlists: :environment do
+    dry_run = is_dry_run?
+    count = 0
+    SubscriberList.find_each do |l|
+      if l.subscriptions.count.zero?
+        count += 1
+        l.destroy unless dry_run
```
Comment (bilbof) [inline lib/tasks/clean.rake]: I wonder if this could be added as a class in the `Clean` module, so we could add tests for it?

**Comment by cbaines** [comment]
What's the motivation for this change? The commit message is a bit lacking.

Also, is there any data related to the subscriber lists in the Email Alert API database that's going to block deleting subscriber lists, or be deleted along with them?

**Comment by bilbof** [comment]
What happens to the subscriptions that reference the deleted subscriber lists?

**Comment by barrucadu** [comment]
@cbaines Creating the `SubscriberList` happens before the user actually enters their email address.  We also don't do any validation on facet values (there's a separate card for that), so as a result we've got a bunch of `SubscriberList`s for organisation with names like `"+response.write(9205521*9921377)+"` which don't have any subscribers.  So it's just to clean those up.

@bilbof It's only deleting `SubscriberList`s with no `Subscription`s, so I don't think there will be any?

**Comment by bilbof** [comment]
@barrucadu 🤦‍♂ that's fine then 😄

File: spec/lib/clean/empty_subscriber_lists_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  context "when there are lists with subscribers" do
+    let!(:list) { create(:subscriber_list_with_subscribers) }
+
+    describe "#lists" do
+      it "returns zero subscriber lists" do
+        expect(subject.lists.count).to eq(0)
+        expect(subject.lists).to_not include(list)
+      end
+    end
```
Comment (bilbof) [inline spec/lib/clean/empty_subscriber_lists_spec.rb]: Thanks for adding specs. Looks great!

Could you add a test for `remove_empty_subscriberlists`?

Comment (barrucadu) [reply spec/lib/clean/empty_subscriber_lists_spec.rb]: Done

**Pull request merged**